tls/tests/test_local.py

```diff
@@ -0,0 +1,171 @@
+    c.check(None)
+
+    aggregator.assert_service_check(c.SERVICE_CHECK_CAN_CONNECT, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VERSION, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VALIDATION, status=c.OK, tags=c._tags, count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_EXPIRATION, status=c.OK, tags=c._tags, count=1)
+
+    aggregator.assert_metric('tls.days_left', count=1)
+    aggregator.assert_metric('tls.seconds_left', count=1)
+    aggregator.assert_all_metrics_covered()
+
+
+def test_ok_der(aggregator, instance_local_ok_der):
+    c = TLSCheck('tls', {}, [instance_local_ok_der])
+    c.check(None)
+
+    aggregator.assert_service_check(c.SERVICE_CHECK_CAN_CONNECT, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VERSION, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VALIDATION, status=c.OK, tags=c._tags, count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_EXPIRATION, status=c.OK, tags=c._tags, count=1)
+
+    aggregator.assert_metric('tls.days_left', count=1)
+    aggregator.assert_metric('tls.seconds_left', count=1)
+    aggregator.assert_all_metrics_covered()
+
+
+def test_hostname(aggregator, instance_local_hostname):
+    c = TLSCheck('tls', {}, [instance_local_hostname])
+    c.check(None)
+
+    aggregator.assert_service_check(c.SERVICE_CHECK_CAN_CONNECT, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VERSION, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VALIDATION, status=c.OK, tags=c._tags, count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_EXPIRATION, status=c.OK, tags=c._tags, count=1)
+
+    aggregator.assert_metric('tls.days_left', count=1)
+    aggregator.assert_metric('tls.seconds_left', count=1)
+    aggregator.assert_all_metrics_covered()
+
+
+def test_hostname_mismatch(aggregator, instance_local_hostname_mismatch):
+    c = TLSCheck('tls', {}, [instance_local_hostname_mismatch])
+    c.check(None)
+
+    aggregator.assert_service_check(c.SERVICE_CHECK_CAN_CONNECT, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VERSION, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VALIDATION, status=c.CRITICAL, tags=c._tags, count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_EXPIRATION, status=c.OK, tags=c._tags, count=1)
+
+    aggregator.assert_metric('tls.days_left', count=1)
+    aggregator.assert_metric('tls.seconds_left', count=1)
+    aggregator.assert_all_metrics_covered()
+
+
+def test_cert_bad(aggregator, instance_local_cert_bad):
+    c = TLSCheck('tls', {}, [instance_local_cert_bad])
+    c.check(None)
+
+    aggregator.assert_service_check(c.SERVICE_CHECK_CAN_CONNECT, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VERSION, count=0)
+    aggregator.assert_service_check(c.SERVICE_CHECK_VALIDATION, status=c.CRITICAL, tags=c._tags, count=1)
+    aggregator.assert_service_check(c.SERVICE_CHECK_EXPIRATION, count=0)
```

Comment:
Should we also be sending this as CRITICAL? With service checks, the last value will stay in the UI. So in this case you'd get an alert on `SERVICE_CHECK_VALIDATION` but not on `SERVICE_CHECK_EXPIRATION `. Which may be ok, just wanted to mention it.

Reply:
I think not as that would be misleading since parsing errors would make expiry check impossible

Reply:
Totally fair.